cognition_objects/message.py

```diff
@@ -21,6 +22,24 @@ def get_all_by_conversation_id(
     )
 
 
+def get_all_by_conversation_ids(
+    project_id: str, conversation_ids: List[str]
+) -> Dict[str, List[CognitionMessage]]:
+    messages = (
+        session.query(CognitionMessage)
+        .filter(
+            CognitionMessage.project_id == project_id,
+            CognitionMessage.conversation_id.in_(conversation_ids),
+        )
+        .order_by(CognitionMessage.created_at.asc())
+        .all()
+    )
+    messages_by_conversation = defaultdict(list)
+    for message in messages:
+        messages_by_conversation[message.conversation_id].append(message)
```

Comment:
also doable in sql if we want to provide a dict instead of an sql alchemy object.

```
SELECT jsonb_object_agg(conversation_id, messages)
FROM (
	SELECT m.conversation_id, array_agg(row_to_json(M)) messages
	FROM cognition.message m 
	WHERE m.project_id = '72c5721d-6cd4-4652-97b7-3aeca0600e07'
	--AND m.conversation_id IN ('')
	GROUP BY m.conversation_id
) x
```